### ISSUE TITLE
Properties with dots in their name are treated as paths instead of as literals

### ISSUE THREAD

**Issue by VanTanev** [mswjs/data#149]
Given the following code:

```ts
export const db = factory({
  user: {
    'employee.id': primaryKey(() => 'abc-123'),
  }
})

db.user.create({ 'employee.id': 'some-id' })
```

We get this error:
```
Uncaught (in promise) OperationError: Failed to create a "user" entity: expected the primary key "employee.id" to have a value, but got: undefined
    at new OperationError (OperationError.js:28)
    at Object.create (factory.js:66)
```

This is because createModel uses `lodash/set`, which converts dot strings to deep properties (eg, among others):
https://github.com/mswjs/data/blob/c4a5208301660f4abcfa9996ce301c34f5047aa0/src/model/createModel.ts#L58-L65

The resulting model object is:
```ts
  const user: {
    employee: {
      id: <primaryKey>
    }
  }
  // instead of 
  const user: {
    'employee.id': <primaryKey>
  }
```

I think that `lodash/set` is a poor fit, because properties with dots in them are completely valid. Is there any reason it was used other than convenience?

**Comment by VanTanev** [comment]
Hey @kettanaito, I have already implemented this in #152. Is there anything you need me to improve?

**Comment by kettanaito** [comment]
@VanTanev, thank you for your work on that! I've seen the pull request, will review it as soon as I've got a spare minute. Looking forward to having this issue resolved.
